File: Palindrome.java
public class Main {
    static void palindrome(String str) {
        boolean flag = true;
        int i = 0, j = str.length() - 1; 
  
        while (i < j) { 
            if (str.charAt(i) != str.charAt(j)) 
                flag = false; 
            i++; 
            j--; 
        }
        if (flag) {
            System.out.println(str + " is palindrome.");
        } else {
            System.out.println(str + " is not palindrome.");
        }
    }

    public static void main(String[] args) {
        palindrome("Hello");
        palindrome("BOB");
    }
}

File: Palindrome.kt
fun palindrome(str: String) {
    var flag = true
    for (i in 0..(str.length/2-1)) {
        if(str[i] != str[str.length-1-i]) {
            flag = false
            break
        }
    }
    if (flag == true) {
        println("$str is palindrome.")
    } else {
        println("$str is not palindrome.")
    }
}

fun main(args: Array<String>) {
    palindrome("Hello")
    palindrome("BOB")
}
